build/source/dshare/data_types.f90

```diff
@@ -0,0 +1,140 @@
+ USE multiconst,only:integerMissing
+ implicit none
+ private
+
+ ! ***********************************************************************************************************
+ ! Define the model decisions
+ ! ***********************************************************************************************************
+ ! the model decision structure
+ type,public  :: model_options
+  character(len=64)                      :: cOption='notPopulatedYet'
+  character(len=64)                      :: cDecision='notPopulatedYet'
+  integer(i4b)                           :: iDecision=integerMissing
+ end type model_options
+
+ ! ***********************************************************************************************************
+ ! Define metadata for model forcing datafile
+ ! ***********************************************************************************************************
+ ! define a derived type for the data in the file
+ type,public  :: file_info
+  character(len=256)                     :: filenmDesc='notPopulatedYet' ! name of file that describes the data
+  character(len=256)                     :: filenmData='notPopulatedYet' ! name of data file
+  integer(i4b)                           :: ncols                    ! number of columns in the file
+  integer(i4b)                           :: ixFirstHRU               ! index of the first HRU to share the same data
+  integer(i4b),allocatable               :: time_ix(:)               ! column index for each time variable
+  integer(i4b),allocatable               :: data_ix(:)               ! column index for each forcing data variable
+ end type file_info
+
+ ! ***********************************************************************************************************
+ ! Define metadata on model parameters
+ ! ***********************************************************************************************************
+ ! define a data type to store model parameter information
+ type,public  :: par_info
+  real(dp)                               :: default_val              ! default parameter value
+  real(dp)                               :: lower_limit              ! lower bound
+  real(dp)                               :: upper_limit              ! upper bound
+ endtype par_info
+
+ ! ***********************************************************************************************************
+ ! Define variable metadata
+ ! ***********************************************************************************************************
+ ! define derived type for model variables, including name, decription, and units
+ type,public :: var_info
+  character(len=64)                      :: varname=''       ! variable name
+  character(len=128)                     :: vardesc=''       ! variable description
+  character(len=64)                      :: varunit=''       ! variable units
+  character(len=32)                      :: vartype=''       ! variable type (scalar, model layers, etc.)
```

Comment:
issue #118 - also I would be more generous in allowing longer strings to avoid undersizing strings.